retail/api/grs-search/SearchWithQueryExpansion.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_search_for_products_with_query_expansion_specification]
+// Call Retail API to search for a products in a catalog,
+// enabling the query expansion feature to let the Google Retail Search build an automatic query expansion.
+
+using Google.Cloud.Retail.V2;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace grs_search.search
+{
+    public static class SearchWithQueryExpansion
+    {
+        private const string Endpoint = "retail.googleapis.com";
+        private static readonly string ProjectNumber = Environment.GetEnvironmentVariable("PROJECT_NUMBER");
```

Comment:
Please do not include this on the sample code but on the test code (usually in a fixture). Clients might not have/want to set the project Ids on environment variables.

All fields that a sample uses should be received as parameters for testing and for being able to call the sample method from anywhere.

Reply:
The purpose of this code samples is dual: first, they will be used as part of the documentation, here https://cloud.google.com/retail/docs/search-basic#search_product-python.

The second purpose is to run this code as the interactive tutorials, that will be run from the CloudShell. 
In this tutorials, we are asking user to create their own project and set environment variables PROJECT_NUMBER, PROJECT_ID, BUCKET_NAME, after that the user will be able to run the code samples.

For the tests we will use our project, and it's name, should be defined in the Secret Manager. I kindly ask your assistance in adding the values to the Secret Manager